tests/schema/tests.py

```diff
@@ -3515,6 +3516,41 @@ def test_unique_constraint_field_and_expression(self):
             editor.remove_constraint(Author, constraint)
         self.assertNotIn(constraint.name, self.get_constraints(table))
 
+    @isolate_apps("schema")
+    def test_unique_constraint_composite_pk_in_expressions(self):
+        class Release(Model):
+            pk = CompositePrimaryKey("version", "name")
+            version = IntegerField()
+            name = CharField(max_length=20)
+            release_date = DateField()
+
+            class Meta:
+                app_label = "schema"
+
+        with connection.schema_editor() as editor:
+            editor.create_model(Release)
+        # Add constraint.
+        constraint = UniqueConstraint(
+            F("pk"),
```

Comment:
Should we disallow references to composite fields instead and force their flattening?

e.g.

```suggestion
            F("version"),
            F("name"),
```

It feels like we're opening a can of worms that we should preferably leave closed until we learn more about how these fields are going to be used in the wild. A can of worm that we have the luxury of keeping closed because in the case of `UniqueConstraint` nothing can be more unique than its primary key so it doesn't make much sense to support this in the first place.

Reply:
> ... in the case of `UniqueConstraint` nothing can be more unique than its primary key so it doesn't make much sense to support this in the first place.

It will crash in the same way on a pure `Index`.

Reply:
... but yes, I was also thinking about adding raising an error when composite PK is passed to `fields` or `expressions` (ticket-35997) :shrug:

Reply:
> It will crash in the same way on a pure Index.

Isn't it relatively rare that a primary key column inclusion in a secondary index makes sense?

My concerns with adding support for composite fields references is that it opens the door to support other exotic specifications like

```python
Index("-pk")

Index(OrderBy("pk", nulls_last=True)) 

Index(OpClass("pk", name="varchar_pattern_ops")) 
```

There are a lot of unknowns at this point and we've usually taken a stance of raising not-supported exception at first when landing new features to avoid crashes and give us some time to determine if a feature is worth adding.

I'm not strongly opposed to adding this feature but I wanted to point out that it brings more questions to me than it provides answers.

Reply:
> My concerns with adding support for composite fields references is that it opens the door to support other exotic specifications like

I think we've already (unfortunately) opened this box :bomb: since composite PKs are not prohibited by any expressions or database functions :scream: Nevertheless, I will prepare an alternative PR later today.

Reply:
Check out https://github.com/django/django/pull/18922.